src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3926,8 +3938,45 @@ public static void prohibitPartialHalfSlices(Design design) {
                         bottomUsed = true;
                     }
                 }
+                if (c.getBEL().isFF()) {
+                    String belName = c.getBELName();
+                    char letter = belName.charAt(0);
+                    if (si.getCell(letter + "6LUT") == null && si.getCell(letter + "5LUT") == null) {
```

Comment:
I'm guessing there is a very slim (but non zero) chance that this condition might not be true and we miss the opportunity to insert a RT. 

It would be great to get a warning (even under some verbose setting) emitted that a RT won't be considered here just in case we have to shoot around in the dark. It seems like this could be achieved by deferring this check until later -- first you check whether the node is used, and the existence of a sitewire assignment (does `makeBlackbox()` remove site wires assignments?) or SPI, before attempting to insert a RT.